### ISSUE TITLE
Need better RGB color implementation.

### ISSUE THREAD

**Issue by caternuson** [adafruit/Adafruit_CircuitPython_TCS34725#18]
The current implementation of `color` is in need of replacement. See discussions in #14 and #10 .

**Comment by FoamyGuy** [comment]
@caternuson Is this still something that is needed? Looking at the PR referenced at the bottom of the issues you linked it looks like this library now matches the the code from that PR.

**Comment by caternuson** [comment]
Yep, that PR was merged. But I think this issue is still open. It relates specifically to the current code in the `color` function. More specifically, the odd math being done in `color_rgb_bytes`:
```python
        red   = int(pow((int((r/clear) * 256) / 255), 2.5) * 255)
        green = int(pow((int((g/clear) * 256) / 255), 2.5) * 255)
        blue  = int(pow((int((b/clear) * 256) / 255), 2.5) * 255)
```
I'm not sure the origins of that are understood. Note how nothing else in the library uses `color_rgb_bytes`. The other values, like `lux` and `color_temperature` use `color_raw`.

To comply with the CircuitPython Design Guide, a property named `color` needs to be provided. One idea might be to just use `color_raw` for that as well. I think things stand at sort of stale mate since the reasoning for `color_rgb_bytes` is unknown.

**Comment by FoamyGuy** [comment]
Ah, I understand now. Thank you.

**Comment by kattni** [comment]
@FoamyGuy Are you interested in working on this?

**Comment by FoamyGuy** [comment]
@kattni I will try to tackle it. The details of the color computations are a bit over my head but clicking through some of the linked issues and PR threads I found more information and links to other resources. 

I can start by trying to use the `color_raw` value instead of `color_rgb_bytes` and compare that with the current code. 

It seems the header breakouts for this device are out of stock at Adafruit and Digikey currently. https://www.adafruit.com/product/1334

However there is a Flora variant of the sensor breakout also and there are plenty of those available. I will get one ordered to use for testing the changes.

**Comment by boukeas** [comment]
I am pretty sure the `pow(x, 2.5)` part performs [gamma correction](https://en.wikipedia.org/wiki/Gamma_correction) and the rest are just normalisation and scaling transforms. However, I am not at all sure about the robustness of these operations and I wasn't able to find a source for them.

**Comment by silviaCC** [comment]
In the projects that use the color sensor, like the [Piano Glove](https://learn.adafruit.com/pianoglove/program-it), [FLORAbrella](https://learn.adafruit.com/florabrella/code), and [Chameleon Scarf](https://learn.adafruit.com/chameleon-scarf/code), the Arduino code given uses the raw values from the sensor, normalizes them, and then applies a gamma correction. 

I'm pretty sure this is equivalent to the calculations in this issue. 

![gamma transform](https://user-images.githubusercontent.com/4115613/123839093-05cc2800-d90d-11eb-99ff-0e7c96c6bec0.png)

From what I can tell, the code in `color_rgb_bytes` runs correctly, and there is nothing much to do, except maybe add some comments explaining the calculations done. 

I would also suggest adding some extra code to `tcs34725_simpletest.py `, because at the moment it's not actually showing how to use the sensor to read a color, just the color temperature.

**Comment by kattni** [comment]
@FoamyGuy @caternuson Do you mind following up with @silviaCC's suggestion regarding this issue? Thanks!

**Comment by kattni** [comment]
@silviaCC Please file another issue for the simpletest update, and if you're interested in doing so, please update the simpletest.py file and submit a PR.

**Comment by silviaCC** [comment]
@kattni  Sure, I can do that.

**Comment by caternuson** [comment]
@silviaCC Excellent sleuthing. @boukeas Seems like you also suggested same.

That does appear to be a baked in gamma correction. @kattni I agree with the general idea of closing this issue as well as PR #31. Adding some code comment would be a good idea. @silviaCC Do you want to do that as part of PR to update simpletest?

**Comment by silviaCC** [comment]
@caternuson  Yes, no problem, I'm working on it  (though rather slowly, I'm navigating my first GitHub contribution)

**Comment by jposada202020** [comment]
Fixed by #37 . Thanks @elpekenin for the heads up :)
